### PR TITLE
Breaking: Return pointers from `execute_on_qvm` and `execute_on_qpu`

### PR DESCRIPTION
**Pull request by kalzoo**
TODO:

- [x] `execute_on_qvm`
- [x] `free_execution_result`
- [x] Fix `ANCHOR` tags in `libqcs.h`

### CODE REVIEW

**Review by dbanty — APPROVED** [review]
I think this is ready to go @kalzoo but I've made some tweaks, so please give it a look over before we merge. Also, we should probably squash since this is a bit messy now. I set the title to what I think the message should be.

**Comment by dbanty** [comment]
Actually looks like semrel [is broken](https://github.com/rigetti/qcs-sdk-c/runs/4412073626?check_suite_focus=true) so let's not merge until I get a chance to figure out why

**Comment by kalzoo** [comment]
@dbanty thank you for polishing this up

**Pull request merged**
